### ISSUE TITLE
DISABLED test_cast_round_trip_soak_cuda_float8_e4m3fn (__main__.TestFloat8DtypeCUDA)

### ISSUE THREAD

**Issue by huydhn** [pytorch/pytorch#113829]
Platforms: linux, win

This test was disabled because it is failing on main branch ([recent examples](https://hud.pytorch.org/failure?name=periodic%20%2F%20linux-focal-cuda11.8-py3.10-gcc9-debug%20%2F%20test%20(default%2C%203%2C%205%2C%20linux.4xlarge.nvidia.gpu)&jobName=undefined&failureCaptures=%5B%22test_quantization.py%3A%3ATestFloat8DtypeCUDA%3A%3Atest_cast_round_trip_soak_cuda_float8_e4m3fn%22%5D)).

This test is failing on periodic debug build with a `Tensor-likes are not equal` error, probably because of this change https://github.com/pytorch/pytorch/pull/107586

cc @galexite @malfet

**Comment by galexite** [comment]
I don't think it's likely to be #107586, I think it's likely to be the same failure as before (from #113361) but the test has been renamed or moved since (from `test_float8.py` to `test_quantization.py`.

**Comment by malfet** [comment]
OK, I should have been more careful with code reviews: one should never compare floats with `atol=0` as order of operations/compilation flags can have effect on the accuracy

**Comment by malfet** [comment]
I.e. here is a fix:
```
diff --git a/test/quantization/core/experimental/test_float8.py b/test/quantization/core/experimental/test_float8.py
index 54d45f905b1..fb4a23b1d3c 100644
--- a/test/quantization/core/experimental/test_float8.py
+++ b/test/quantization/core/experimental/test_float8.py
@@ -219,7 +219,7 @@ class TestFloat8Dtype(TestCase):
         x = torch.cat((x, -x))
         x8 = x.to(dtype)
         x8_simulated = simulate_fp8_precision(x, dtype)
-        self.assertEqual(x8_simulated, x8.float(), atol=0, rtol=0)
+        self.assertEqual(x8_simulated, x8.float())
 
     @dtypes(*FLOAT8_DTYPES)
     @dtypesIfCUDA(*CUDA_FLOAT8_DTYPES)
```
